File: pset1/greedy.c
/**
 * Pset 1
 * CS50
 * Mohammed Attya
 *
 * @GitHub mohammedattya2
 *
 * MIT Licence
 *
 * OpenSource OpenMind
 *
 */

#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main (void)
{
    float owed = 0;

    do
    {
        printf("hai! How much change is owed?\n");
        owed = get_float();
    } while (owed < 0);
    int counter = 0;

    int change = (int) round(owed * 100);

    while (change > 0)
    {
        if (change >= 25)
        {
            change -= 25;
            counter++;
        }
        else if (change >= 10)
        {
            change -= 10;
            counter++;
        }
        else if (change >= 5)
        {
            change -= 5;
            counter++;
        }
        else
        {
            change -= 1;
            counter++;
        }
    }
    printf("%i\n", counter);

}

File: pset6/greedy.py
"""
CS50

Problem set6

Mohammed Attya

"""

while True:
    change = input('hai! How much change is owed? ')
    try:
        change = float(change) * 100
    except Exception as e:
        print('Retry: ...')
    else:
        break

if change < 0:
    print('change must be > 0')
    exit(1)

counter = 0
while change > 0:
    if change >= 25:
        change -= 25
        counter += 1

    elif change >= 10:
        change -= 10
        counter += 1

    elif change >= 5:
        change -= 5
        counter += 1

    else:
        change -= 1
        counter += 1
print(counter)
